firmware/library/L1_Drivers/dac.hpp

```diff
@@ -1,19 +1,209 @@
+    84, 89, 94, 99, 104, 109, 114, 119, 124, 130,
+    135, 141, 147, 152, 158, 164, 170, 176, 182, 188, 194,
+    200, 206, 212, 218, 225, 231, 237, 243, 250, 255
+    };
+    // PinConfigure led3(PinConfigure::CreatePinConfigure<1, 15>());
+    /* Initialize Dac by setting the clock divider and enabling 
+      The Dac Pin withiin IOCON.  Then set the initial Bias level to 0 */
+    void Initialize(void) override
+    {
+        // Set Clock Divider to 1
+        LPC_SC->PCLKSEL = (1 << kClockDividerReg);
+        // Enable DAC Pin P0.26
+        LPC_IOCON->P0_26 = (1 << kEnableDacReg);
+
+        EnableDAC(true);
+        // Set Update Rate to 1MHz
+        LPC_DAC->CR &= ~(1 << kBiasReg);
+        // return true;
+    }
+    // The Dac ouptut is a 10 bit input and thus it is necessary to
+    // ensure dac_output is less than 1024(largest 10-bit number)
+    bool WriteDac(uint16_t dac_output) override
+    {
+        if ( dac_output < 1024 )
+        {
+            LPC_DAC->CR = (dac_output << kDacOutReg);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+    // Takes an input voltage and converts the float value and calculates
+    // the conversion necessary and then typecasts it to an integer.
+    // if the voltage value is greater than 3.3 it will fail
+    bool SetVoltage(float voltage) override
+    {
+        // value = (dac_out * 1024)/VrefP
+        float value = (voltage * 1024) / kVrefP;
+        int   conversion = static_cast<int>(value);
+        if ( voltage < 3.3 )
+        {
+            LPC_DAC->CR = (conversion << kDacOutReg);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+    // Demonstration Function for Sine demo,  Still need to determine
+    // the correct way to incorperate frequency into the output
+    void SineDac(uint32_t frequency_hz) override
+    {
+        for ( uint8_t i = 0; i < 255; i++ )
+        {
+            LPC_DAC->CR = (kSineLookup[i] << kDacOutReg);
+            // TODO(Conrad) Determine wait time based on frequency and testing
+        }
+    }
+    // Demonstration Function for Sawtooth demo,  Still need to determine
+    // the correct way to incorperate frequency into the output
+    void SawtoothDac(uint32_t frequency_hz) override
+    {
+        for ( uint8_t i = 0; i < 255; i++ )
+        {
+            LPC_DAC->CR = (kSawtoothLookup[i] << kDacOutReg);
+            // TODO(Conrad) Determine wait time based on frequency and testing
+        }
+    }
+    // Demonstration Function for Triangle demo,  Still need to determine
+    // the correct way to incorperate frequency into the output
+    void TriangleDac(uint32_t frequency_hz) override
+    {
+        for ( uint8_t i = 0; i < 255; i++ )
+        {
+            LPC_DAC->CR = (kTriangleLookup[i] << kDacOutReg);
+            // TODO(Conrad) Determine wait time based on frequency and testing
+        }
+    }
```

Comment:
I think these demonstration methods should only be apart of the example project and not the class.

Reply:
Ok I have moved them to my main for example project.  Should the main be code reviewed as well?  Its not going to be pulled into the repository is it?